config/opal_check_libnl.m4

```diff
@@ -0,0 +1,228 @@
+dnl -*- shell-script -*-
+dnl
+dnl Copyright (c) 2015-2016 Research Organization for Information Science
+dnl                         and Technology (RIST). All rights reserved.
+dnl $COPYRIGHT$
+dnl
+dnl Additional copyrights may follow
+dnl
+dnl $HEADER$
+dnl
+
+dnl
+dnl Helper function to parse --with-libnl* options
+dnl
+dnl $1: variable name
+dnl $2: --with-<foo> value
+dnl
+AC_DEFUN([OPAL_LIBNL_PARSE_WITH],[
+	case "$2" in
+	no)
+		# Nope, don't want it
+		opal_want_$1=no
+		;;
+	yes)
+		# Yes, definitely want it
+		opal_want_$1=yes
+		;;
+	default)
+		# Default case -- try and see if we can find it
+		opal_want_$1=default
+		opal_$1_location=/usr
+		;;
+	*)
+		# Yes, definitely want it -- at a specific location
+		opal_want_$1=yes
+		opal_$1_location="$2"
+		;;
+	esac
+])
+
+dnl OPAL_LIBNL_SANITY_INIT()
+dnl --------------------------------------------------------------------
+AC_DEFUN([OPAL_LIBNL_SANITY_INIT], [
+    opal_libnl_version=0
+    opal_libnl_libs=
+    opal_libnl3_libs=
+    AC_ARG_WITH([libnl],
+                [AC_HELP_STRING([--with-libnl(=DIR)],
+                                [Directory prefix for libnl (typically only necessary if libnl is installed in a location that the compiler/linker will not search by default)])],
+                [], [with_libnl=default])
+
+    # The --with options carry two pieces of information: 1) do
+    # you want a specific version of libnl, and 2) where that
+    # version of libnl lives.  For simplicity, let's separate
+    # those two pieces of information.
+    OPAL_LIBNL_PARSE_WITH([libnl], [$with_libnl])
+])
+
+dnl OPAL_LIBNL_SANITY_CHECK(lib, function, LIBS)
+dnl --------------------------------------------------------------------
+AC_DEFUN([OPAL_LIBNL_SANITY_CHECK], [
+    case $host in
+        *linux*)
+            AC_LANG_PUSH(C)
+            cat > conftest_c.$ac_ext << EOF
+extern void $2 (void);
+int main(int argc, char *argv[[]]) {
+    $2 ();
+    return 0;
+}
+EOF
+            OPAL_LOG_COMMAND([$CC -o conftest $CFLAGS $CPPFLAGS conftest_c.$ac_ext $LDFLAGS -l$1 $LIBS $3],
+                             [ldd_output=`ldd conftest`
+                              libnl_version=0
+                              AS_IF([echo $ldd_output | grep -q libnl.so],
+                                    [AS_IF([test $opal_libnl_version -eq 3],
+                                           [OPAL_UNIQ([opal_libnl3_libs])
+                                            AC_MSG_WARN([lib nl version conflict: $opal_libnl3_libs requires libnl-3 whereas $1 requires libnl])],
```

Comment:
Is there a reason we only warn here (and don't error/abort)?  Is the intent to go through and find *all* conflicts so that they can be reported at the end?  If so, it might be worth a comment (or another AC_MSG_WARN?) explaining this.

(same question applies a few lines down where we warn again [vs. abort])